packages/hash/api/src/graphql/typeDefs/knowledge/page.typedef.ts

```diff
@@ -0,0 +1,232 @@
+import { gql } from "apollo-server-express";
+
+export const knowledgePageTypedef = gql`
+  type KnowledgePage implements KnowledgeEntity {
+    contents: [KnowledgeBlock!]!
+    archived: Boolean
+    summary: String
+    title: String!
+    icon: String
+
+    # ENTITY INTERFACE FIELDS BEGIN #
+    """
+    The id of the entity
+    """
+    entityId: ID!
+    """
+    The specific version of the entity
+    """
+    entityVersionId: String!
+    """
+    The id of the account that owns this entity.
+    """
+    ownedById: ID!
+    """
+    The fixed id of the type this entity is of.
+    """
+    entityTypeId: ID!
+    """
+    The full entity type definition.
+    """
+    entityType: PersistedEntityType!
+    """
+    The linked entities of the entity.
+    """
+    linkedEntities: [KnowledgeEntity!]!
+    """
+    The JSON object containing the entity's properties.
+    """
+    properties: JSONObject!
+    # ENTITY INTERFACE FIELDS END #
+  }
+
+  type KnowledgeEntityRef {
+    accountId: ID!
+    entityId: ID!
+    entityVersion: String!
+  }
+
+  """
+  Insert a block into a page with a corresponding entity.
+  """
+  input KnowledgeInsertBlockAction {
+    """
+    The account ID to create the block and associated entity in.
+    """
+    accountId: ID!
```

Comment:
I think it's inconsistent for `KnowledgeBlock`/`KnowledgePage` to have `ownedById`, and for these methods to require `accountId`. I'm not sure exactly what to do about it though.

We could start using `ownedById` in the new GQL types, but [as @Alfred-Mountfield pointed out](https://hashintel.slack.com/archives/C03F7V6DU9M/p1663233718879279) this will worsen the diff we have to maintain between `dev/graph` and `main`.

Perhaps a temporary solution to keep things a little more consistent in GQL would be to have both `accountId` and `ownedById` fields on the GQL types, where `accountId` is an alias of `ownedById`. In mutations we would have to continue using `accountId`. What do you think?

cc @Alfred-Mountfield

Reply:
I think we should make use of `ownedById` here as it's a new field that we control right now. As we port frontend to the new resolvers, we will have to change the usages incrementally, so I don't fear cascading updates

Reply:
I changed it in 58b4445818807f601e27d8fe1ae042c70d46b8c0

Reply:
The `accountId` alias in the types may still be a good idea though. Without it any mutation we replace in the FE will have a return type that will be incompatible with the prior types otherwise, and the changes would cascade across the FE which I think we should try and avoid.

Reply:
That is a good point. Perhaps we should add that idea to https://app.asana.com/0/1200211978612931/1202986802899184/f

Reply:
I don't think that should be the task as I believe it was envisioned for post-merge into `main`, whereas what I'm discussing should happen before merge into `main` (as that's when we would benefit from the alias).